### DISCUSSION TITLE
shouldn't atom set  updater called multiple times ?

### DISCUSSION THREAD

**Discussion by dante01yoon** [pmndrs/jotai#1718] [Q&A] [answered]
Greetings,

Is atom setter use React's setState internally?
shouldn't call atom setter multiple times in single updater called?
### 1.
```ts
const example = atom(null, (get, set, pair) => {
   pair.forEach((item) => {
       set(sameAtom, item)
   })
   
})
```

or

### 2. 
```ts
const exteriorAtom = atom(0);
atom(null, (get, set) => {
  const prevCount = get(exteriorAtom);
  set(exteriorAtom,  prevCount + 1);
   set(exteriorAtom,  prevCount + 1);
})
```

or 
### 3.
```ts
const [ssn, setter] = useAtom(ssnAtom);

useEffect(() => {
   setter({ ...ssn, name: "jotai"})
   setter({ ...ssn, id: "something"}); 
},[]) 
```

all of above situations are not recommended like using same setState in React right?

**Comment by dai-shi** [comment, marked as answer]
Right, but not because how React works. It's because how JS works. It has nothing to do with jotai nor react.

**Reply by dante01yoon** [reply]
are you talking about asynchronous and closure ?

**Reply by dai-shi** [reply]
Thought closure issue, not async.
